### ISSUE TITLE
Texture Rect size doesn't go below image size

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot#13179]
**Godot version**
Godot 3 alpha 2

**Issue description:**
TextureRect doesn't scale below image size in scale stretch mode. The only way to scale it is to directly use the scale property and edit the values.

**Proposal**
Let the texture frame go below the image size and have different handles for margin change and scale change.

**Comment by reduz** [comment]
This is not a bug, it's working as intended. If you want to make it shrink less than it's size then toggle the "expand" property. Otherwise, this would be unusable in containers.

**Comment by Soaku** [comment]
Toggling "expand" doesn't work for me. 😕

**Comment by marineorganism** [comment]
`var arvr_interface = ARVRServer.find_interface("Native mobile")
if arvr_interface and arvr_interface.initialize():
    get_viewport().arvr = true`

**Comment by andrew-wilkes** [comment]
Here is some code to downsize an image texture:
func get_resized_texture(t: Texture, width: int = 0, height: int = 0):
	var image = t.get_data()
	if width > 0 && height > 0:
		image.resize(width, height)
	var itex = ImageTexture.new()
	itex.create_from_image(image)
	return itex

**Comment by JoelPagliuca** [comment]
Just bumped into this same behavior and the TextureProgress no longer has an `expand` property as far as I can tell, has something changed?

**Comment by Calinou** [comment]
> Just bumped into this same behavior and the TextureProgress no longer has an expand property as far as I can tell, has something changed?

No, you can scale the Control node if you really need it to display with a smaller size.

**Comment by JoelPagliuca** [comment]
yeah just was curious because this behavior is a little odd (I struggle with UI elements so maybe just me), thanks @Calinou !

**Comment by JoelPagliuca** [comment]
I'm also noticing in my scripting the `rect_size` of the `TextureProgress` is what I've set in the UI even after programatically setting the `texture_progress` to a larger image. So the workaround I'm using here is setting a `texture_progress` to a dummy image in the UI, which now means all my textures for this control in the future have to be this same size

**Comment by DoctorPepper** [comment]
To anyone who runs into this issue in the future, here's the recipe I just discovered through trial and error:

`Expand: On`
`Stretch Mode: Scale`
then change the `Min Size` to be your desired size. Not sure if this works on TextureProgress, but it does work on other textures!

**Comment by JWR26** [comment]
> To anyone who runs into this issue in the future, here's the recipe I just discovered through trial and error:
> 
> `Expand: On` `Stretch Mode: Scale` then change the `Min Size` to be your desired size. Not sure if this works on TextureProgress, but it does work on other textures!

There appears to be multiple ways round this. I also found that setting the strech mode to "keep aspect" works. Also toggling off the fill & expand size flags helps if you want to go smaller

**Comment by Dunkhan** [comment]
This needs to be reopened,  working as intended is not an acceptable answer. I spent 90 minutes changing every possible value until I finally got this working. In the process I read dozens of threads on reddit and the godot forums where people were at a loss for a solution or used convoluted and hacky workarounds. The godot discord has countless people asking how to do this over and over.

If I set the size of a TextureRect in code it should resize to that size, the fact that it doesn't is either a bug, or a failure of the documentation to describe the correct way of doing this.
